substrate/frame/preimage/src/lib.rs

```diff
@@ -122,7 +122,9 @@ pub mod pallet {
 		type ManagerOrigin: EnsureOrigin<Self::RuntimeOrigin>;
 
 		/// A means of providing some cost while data is stored on-chain.
-		type Consideration: Consideration<Self::AccountId>;
+		///
+		/// Should never return a `None`, implying no cost for a non-empty preimage.
```

Comment:
Why should it not do that? The implementor of the `Consideration` should be able to return as they see fit.  
Otherwise it would probably need a `Consideration` and `MaybeConsideration` trait to disambiguate.

Reply:
Yes, I agree. But I did now want to change the behaviour of this pallet, and make only a patch. Otherwise we need a migration for the preimage pallet.
Maybe I should rephrase it, that we do expect some cost for any non zero footprint, so it should not return none.